core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -676,8 +681,12 @@ public AttachDetachStatus attachPartition(long timestamp, long partitionSize) {
         boolean validateDataFiles = partitionSize < 0;
 
         boolean checkPassed = false;
+        boolean isSoftLink;
```

Comment:
it is declared there because it is used in two try blocks. Initialised and used in the first, used in the second. The initialisation happens before the detachedPath is trimmed